instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Grape
+      # Handles the events instrumented with ActiveSupport notifications.
+      # These handlers contain all the logic needed to create and connect spans.
+      class EventHandler
+        class << self
+          # Handles the start of the endpoint_run.grape event (the parent event), where the context is attached
+          def endpoint_run_start(_name, _id, payload)
+            name = span_name(payload[:endpoint])
+            span = tracer.start_span(name, attributes: run_attributes(payload), kind: :server)
+            token = OpenTelemetry::Context.attach(OpenTelemetry::Trace.context_with_span(span))
+
+            payload.merge!(__opentelemetry_span: span, __opentelemetry_ctx_token: token)
+          end
+
+          # Handles the end of the endpoint_run.grape event (the parent event), where the context is detached
+          def endpoint_run_finish(_name, _id, payload)
+            span = payload.delete(:__opentelemetry_span)
+            token = payload.delete(:__opentelemetry_ctx_token)
+            return unless span && token
+
+            handle_payload_exception(span, payload[:exception_object]) if payload[:exception_object]
+
+            span.finish
+            OpenTelemetry::Context.detach(token)
```

Comment:
I ended up removing this code, so I didn't implement this. Since now the code reuses the Rack span, I don't know if we need this somewhere, but please let me know if we do 🙏